### PR TITLE
Support NuGet versions and expansion

### LINKED ISSUE #229: Version expansion for NuGet
NuGet is *almost* semver, but not exactly: https://docs.microsoft.com/en-us/nuget/concepts/package-versioning#where-nugetversion-diverges-from-semantic-versioning

### PR DESCRIPTION
**Pull request by oliverchang**
Fixes #229.

### CODE REVIEW

File: lib/osv/nuget.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+# https://docs.microsoft.com/en-us/nuget/concepts/package-versioning
+#  - Optional 4th component (x.y.z.R).
+#  - Prerelease components are compared case insensitively.
+#  - Non-major version segments are optional. e.g. "1" is a valid version
+#    number.
+
+
+def _extract_revision(str_version):
+  """Extract revision (4th component) from version number (if any)."""
+  pattern = re.compile(r'^(\d+)(\.\d+)(\.\d+)(\.\d+)(.*)')
```
Comment (inferno-chromium) [inline lib/osv/nuget.py:32]: nit: maybe add example in a comment or use named groups.

**Review by inferno-chromium — APPROVED** [review]

File: lib/osv/nuget.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+"""NuGet version parser."""
```
Comment (jonathanmetzman) [inline lib/osv/nuget.py:1]: This should be below the license.

File: lib/osv/nuget_test.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
+"""NuGet version parser tests."""
```
Comment (jonathanmetzman) [inline lib/osv/nuget_test.py:1]: This should be below the license.

**Review by jonathanmetzman — APPROVED** [review]
LGTM

File: lib/osv/nuget.py (hunk 2 above)
Comment (oliverchang) [reply lib/osv/nuget.py]: done!

File: lib/osv/nuget_test.py (hunk 3 above)
Comment (oliverchang) [reply lib/osv/nuget_test.py]: done!

**Pull request merged**